### DISCUSSION TITLE
MassFlow base unit

### DISCUSSION THREAD

**Discussion by tobiasfrick** [angularsen/UnitsNet#1295] [Q&A] [answered]
Hi all,

Is there a reason why the base unit of `MassFlow` is `GramPerSecond`, and not `KilogramPerSecond`?
https://github.com/angularsen/UnitsNet/blob/987e83d82cd532b176e9d7ffbc92f50693eb1c3c/Common/UnitDefinitions/MassFlow.json#L3

According to the SI standard the base unit of mass is kilogram, and the other mass-related units all use kilogram for their base unit (e.g. Density -> KilogramPerCubicMeter, MassFlux -> KilogramPerSecondPerSquareMeter, ...)

If there isn't a specific reason for this, I could prepare a pull request and change it.
Thanks!

**Comment by angularsen** [comment, marked as answer]
Hi, the reason is to allow using `Prefixes` array in json to define MilligramPerSecond, KilogramPerSecond, etc.

The name is a bit misleading, it would be better named `ConversionBaseUnit` and could very well be made `private` since it is more an implementation detail than something consumers should rely on for much.

`BaseUnits` in JSON is used to map units to their SI base unit representation. For example, `LengthUnit.Foot` has `"BaseUnits": { "L": "Foot" }`, which can be used to choose a consistent unit representation across quantities.

Some more details here:
https://github.com/angularsen/UnitsNet/wiki/Adding-a-New-Unit#baseunits-optional---the-si-base-units-of-a-unit

Hope this helps!

**Reply by tobiasfrick** [reply]
Understood, thanks for the answer!
My use case is that I have measurements of different quantities that are all given in SI base units, and I wanted to use the base unit as 'source' unit for conversions.
But in that case I won't use it for that and specify the source unit separately.
